NAS-134494 / 25.10 / Add user picker component

denysbutenko opened the pull request:
**Changes:**

Added `ix-user-picker` component with ability to create new users.

**Testing:**

Use the `API Key form` on the `My API Keys` page. Try to create a new user. Ensure you add a group with privileges to the user (e.g., `builtin_administrators`).

Known issues:
- Duplicate entries on scroll.
- Empty list when no priveledges group is added.

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-134494

AlexKarpov98 reviewed (commented):
Overall looks good.
But:
- Tests failed
- When I create new user as you described I don't see the full list of users, but only 
`Add New & New user created`. Not sure if it's good for the best UX.
<img width="459" alt="Screenshot 2025-04-02 at 10 55 18" src="https://github.com/user-attachments/assets/e1e2e7a7-521b-41e2-9cee-579087ccbfb8" />

Known issues confirmed.

denysbutenko commented:
> Overall looks good.
> 
> But:
> 
> - Tests failed
> - When I create new user as you described I don't see the full list of users, but only 
> 
> `Add New & New user created`. Not sure if it's good for the best UX.
> 

This is API Key specific case depends on Query Params passed to ix-user-picker. So it expected to behave like this.

> <img width="459" alt="Screenshot 2025-04-02 at 10 55 18" src="https://github.com/user-attachments/assets/e1e2e7a7-521b-41e2-9cee-579087ccbfb8" />
> 
> 
> 
> Known issues confirmed.
> 
> 
> 
> 

Remaining fixes are in progress. Thanks!

denysbutenko commented:
Changes:

1. Added hint to inform users. The form won't allow the creation of users with no privileges in the future.
2. Fixed duplicates on scroll.
3. The failed test case was replaced with a to-do item to add it back.

bugclerk commented:
This PR has been merged and conversations have been locked.
If you would like to discuss more about this issue please use our forums or raise a Jira ticket.
